### PR TITLE
Moon Boots

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Adds moon boots, a fun research item that makes the wearer move as if they are in zero gravity.

They can be researched in Localized Magnetism t1 tech alongside magboots.

## Why / Balance

Basically just serves as a fun low-level reward item for science. It can enable some interesting interactions with moths and potential movement tech but it's just a fairly weak item that pads out the science tree.

## Technical details

adds a new IsWeightlessEvent to GravitySystem and sets up magboots to go through that codepath.

Also moves the rest of magboots code into shared since it was basically 99% there.

## Media

image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added Moon Boots! This science research item allows you to move like you're in zero-gravity.

### CODE REVIEW

**Review by Partmedia — APPROVED** [review]
I tested this (and the magboots to make sure they still work) and this seems fine.

This diff is a lot bigger than I expected it to be because it moves some stuff to shared and makes some other minor refactoring things. You know how I feel about combining those :)

Overall I wonder why IsWeightless needs to be an event, instead of having the new anti-gravity clothing system add and remove the existing `MovementIgnoreGravityComponent` to the wearer.

**Comment by EmoGarbage404** [comment]
> Overall I wonder why IsWeightless needs to be an event, instead of having the new anti-gravity clothing system add and remove the existing MovementIgnoreGravityComponent to the wearer.

i generally find using event subscriptions a lot less buggy than dealing with adding/removing components. It avoids issues of conflicting entities trying to add or remove the same component and also doesn't have to support lifetime tracking of any kind.

File: Resources/Prototypes/Entities/Structures/Machines/lathe.yml [hunk 1]
```diff
@@ -707,7 +708,7 @@
     staticRecipes:
       - BoxLethalshot
       - BoxShotgunFlare
-      - BoxShotgunPractice
```
Comment (Partmedia) [inline Resources/Prototypes/Entities/Structures/Machines/lathe.yml:710]: Also pointyhat for changing random whitespace and making me wonder why a boots PR is changing shotgun shells

**Comment by Holinka4ever** [comment]
>It can enable some interesting interactions with moths

Did we just a speed merge an itemn, that you can easily research and craft, that one race benefits from?
Didnt we know that people gonna powergame these aspect for own sake?
These gonna givw a lot of headache to be honest....

**Comment by Partmedia** [comment]
> Did we just a speed merge an item, that you can easily research and craft, that one race benefits from? Didnt we know that people gonna powergame these aspect for own sake? These gonna give a lot of headache to be honest....

You're welcome to state any specific concerns you have about this and we're happy to address them. Otherwise, since the code looked reasonable, worked fine in game, and given how trivial the mechanic is, that it needed further consideration.

**Comment by Holinka4ever** [comment]
> > Did we just a speed merge an item, that you can easily research and craft, that one race benefits from? Didnt we know that people gonna powergame these aspect for own sake? These gonna give a lot of headache to be honest....
> 
> You're welcome to state any specific concerns you have about this and we're happy to address them. Otherwise, since the code looked reasonable, worked fine in game, and given how trivial the mechanic is, that it needed further consideration.

Dunno, its easy to get and make as it seems.
Like on tier 1 and 6 iron is too accessable for moth population to able have the most funky and comfortable movement in station. Its also artificially increase moth player count, due of race benefit against others.

**Comment by Djungelskog2** [comment]
I mean, arent moths still one of the worst races to powergame with? (because of their heat weakness)

**Comment by Aisu9** [comment]
does this make you float above chasm?

**Comment by deltanedas** [comment]
i believe no gravity affects step trigger like that but it could be just on the grid, would need testing

**Pull request merged**